tests/spec_decode/test_multi_step_worker.py

```diff
@@ -477,6 +479,104 @@ def test_multi_step_with_batch_expansion_incorrect_output():
     assert (num_mismatch > 0)
 
 
+@torch.inference_mode()
+@pytest.mark.parametrize('num_steps', [1, 2, 3, 4])
+def test_multi_step_correct_kvcache(num_steps):
+    """Verify that the KV cache of the draft model 
+    is correctly updated for sequences with bonus
+    """
+    seed = 100
+    model_name = "JackFram/llama-68m"
+
+    block_size = 16
+    num_gpu_blocks = 2048 // block_size
+    batch_size = 1
+    multi_step_worker = create_worker(
+        MultiStepWorker,
+        model_name,
+        block_size,
+        num_gpu_blocks,
+        seed,
+        model_runner_cls=TP1DraftModelRunner,
```

Comment:
Yeah actually, I found that for the CI machine, since the backend is XFORMERS instead of FlASH_ATTN, it will go to the else branch [here](https://github.com/vllm-project/vllm/blob/9dc2871edc67577335efcc2bdd02e736cd0fd24f/vllm/spec_decode/multi_step_worker.py#L88) by default because `self.model_runner.supports_gpu_multi_step(expanded_request)` is False (only FLASH_ATTN supports GPU multi step). I guess we can specify the backend explicitly in the tests so that it will hit both branches.